src/read.c

```diff
@@ -4126,9 +4126,10 @@ avifBool avifPeekCompatibleFileType(const avifROData * input)
     if (!avifROStreamReadBoxHeaderPartial(&s, &header, /*topLevel=*/AVIF_TRUE) || memcmp(header.type, "ftyp", 4)) {
         return AVIF_FALSE;
     }
-    if (header.size == 0) {
-        // The size of the 'ftyp' box is 0, which means it goes till the end of the file.
-        // Either there is no brand requiring anything in the file but a FileTypebox (so not AVIF), or it is invalid.
+    if (header.isSizeZeroBox || header.size == 0) {
+        // * If header.isSizeZeroBox is true, the 'ftyp' box goes on till the end of the file. This cannot be valid AVIF since
+        //   there are no other boxes left for the image data.
+        // * If header.size == 0, the 'ftyp' box is empty. So there are no brands identifying as AVIF.
```

Comment:
We let the `header.size == 0` case fail inside the `avifParseFileTypeBox()` call at line 4139 as before. No need to handle it as a special case here, just like we don't handle `header.size` equal to 1, 2, ..., 7 as special cases.

Reply:
Done, i have restored the comment here also to it's previous state.